app/db.py

```diff
@@ -75,57 +84,63 @@ def n_edges(self):
     def node(self, uri):
         # looks up a node by uri (essentially [[wikilink]]).
         # this used to be even worse :)
-        try:
-            node = self.nodes()[uri.lower()]
-            return node
-        except (KeyError, IndexError):
-            # We'll handle 404 in the template, as we want to show backlinks to non-existent nodes.
-            # Return an empty.
-            return Node(uri)
+        # nodes = self.nodes()
+        # try:
+        #     node = nodes[uri.lower()]
+        #     return node
+        # except (KeyError, IndexError):
+        #     # We'll handle 404 in the template, as we want to show backlinks to non-existent nodes.
+        #     # Return an empty.
+        #     return Node(uri)
+        node = query_node(uri.lower())
+        return node
 
     def existing_permutations(self, uri, max_length=4):
         # looks up nodes matching a permutation of the tokenized uri.
-        # 
+        #
         # example use: if [[server-agora]] does not exist, serve [[agora-server]]
         #
         # this is, of course, a terrible implementation and dangerous :)
         # only enable up to 4 tokens as 24 permutations is manageable.
         tokens = uri.split('-')
         permutations = itertools.permutations(tokens, max_length)
         permutations = ['-'.join(permutation) for permutation in permutations]
-        nodes = [node for node in G.nodes(only_canonical=True).values() if node.wikilink in permutations and node.subnodes]
+        nodes = [node for node in G.nodes(only_canonical=True).values(
+        ) if node.wikilink in permutations and node.subnodes]
```

Comment:
This doesn't look nice to me, what's up with the reformatting?

Reply:
probably auto formatter I think it uses pep8?

Reply:
this why I like that gofmt is programmatically enforced. it parses the AST not just line formatting like most language formatters